test/e2e/specs/scaleupdown.go

```diff
@@ -0,0 +1,216 @@
+						},
+						Containers: []corev1.Container{
+							{
+								Name:  sampleDeployment,
+								Image: "openshift/" + sampleDeployment,
+								Ports: []corev1.ContainerPort{
+									{
+										Name:          "http",
+										Protocol:      corev1.ProtocolTCP,
+										ContainerPort: 8080,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		_, err = occli.AppsV1.Deployments(namespace).Create(deployment)
+		Expect(err).NotTo(HaveOccurred())
+		err = wait.PollImmediate(2*time.Second, 1*time.Minute, ready.DeploymentIsReady(occli.AppsV1.Deployments(namespace), sampleDeployment))
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Verifying that the deployment's pods are spread across 2 nodes...")
+		By(fmt.Sprintf("Getting deployment %s.%s", namespace, sampleDeployment))
+		dep, err := occli.AppsV1.Deployments(namespace).Get(sampleDeployment, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(dep).NotTo(BeNil())
+
+		By(fmt.Sprintf("Getting %s.%s deployment's labels", namespace, sampleDeployment))
+		set := labels.Set(dep.Spec.Template.Labels)
+		listOptions := metav1.ListOptions{LabelSelector: set.AsSelector().String()}
+
+		By(fmt.Sprintf("Listing pods matching deployment's labels %+v", set))
+		poditems, err := occli.CoreV1.Pods(namespace).List(listOptions)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(poditems).NotTo(BeNil())
+		pods := poditems.Items
+		Expect(int(dep.Status.ReadyReplicas)).To(Equal(2))
+		Expect(len(pods)).To(Equal(2))
+		nodes := make(map[string]bool)
+		for _, pod := range pods {
+			By(fmt.Sprintf("Found pod %s on node %s", pod.Name, pod.Spec.NodeName))
+			nodes[pod.Spec.NodeName] = true
+		}
+		Expect(len(nodes)).To(Equal(2))
+
+		By("Fetching the scale down manifest")
+		external, err = parseExternalConfig(*update)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(external).NotTo(BeNil())
+
+		By("Calling update on the rp with the scale down manifest")
+		updated, err = azurecli.OpenShiftManagedClusters.CreateOrUpdate(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), *external)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(updated).NotTo(BeNil())
+
+		By("Verifying that the deployment's pods are all on 1 node...")
+		By(fmt.Sprintf("Getting deployment %s.%s", namespace, sampleDeployment))
+		dep, err = occli.AppsV1.Deployments(namespace).Get(sampleDeployment, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(dep).NotTo(BeNil())
+
+		By(fmt.Sprintf("Getting %s.%s deployment's labels", namespace, sampleDeployment))
+		set = labels.Set(dep.Spec.Template.Labels)
+		listOptions = metav1.ListOptions{LabelSelector: set.AsSelector().String()}
+
+		By(fmt.Sprintf("Listing pods matching deployment's labels %+v", set))
+		poditems, err = occli.CoreV1.Pods(namespace).List(listOptions)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(poditems).NotTo(BeNil())
+		pods = poditems.Items
+		Expect(int(dep.Status.ReadyReplicas)).To(Equal(2))
```

Comment:
Instead of checking the deployment status the second time around, you could just check the status of the pods.